timesketch/lib/datastores/opensearch.py

```diff
@@ -1271,3 +1296,505 @@ def flush_queued_events(self, retry_count=0):
+                        while not stop_event.is_set():
+                            try:
+                                output_queue.put(
+                                    event_data_to_export, timeout=1
+                                )  # Short timeout for responsiveness
+                                break  # Item put successfully
+                            except queue.Full:
+                                # Queue is full, wait a bit and retry.
+                                # stop_event is checked at the start of this inner loop.
+                                time.sleep(0.1)  # Avoid busy-waiting
+                            except (  # pylint: disable=broad-exception-caught
+                                Exception
+                            ) as e_q:
+                                log_msg = "[Slice %s/%s] Error putting to queue: %s"
+                                os_logger.error(
+                                    log_msg,
+                                    log_slice_id,
+                                    num_slices,
+                                    str(e_q),
+                                    exc_info=True,
+                                )
+                                # Critical error with queue, signal stop
+                                stop_event.set()
+                                break  # Break from inner for hit in hits loop
+                        if stop_event.is_set():
+                            break  # Break from outer for hit in hits loop
+
+                if stop_event.is_set():
+                    # Break from while True if stop_event was set during item
+                    # processing
+                    break
+
+                total_docs_in_slice += len(hits)
+                os_logger.info(
+                    (
+                        "[Slice %s/%s] Retrieved %s docs. "
+                        "Total for this slice: %s. Approx. queue size: %s"
+                    ),
+                    log_slice_id,
+                    num_slices,
+                    len(hits),
+                    total_docs_in_slice,
+                    output_queue.qsize(),
+                )
+                search_after_params = hits[-1].get("sort")
+                if (
+                    search_after_params is None and hits
+                ):  # Check hits because if no hits, search_after_params is not needed
+                    os_logger.error(
+                        (
+                            "[Slice %s/%s] 'sort' field missing in last hit. "
+                            "Cannot continue with search_after for this slice."
+                        ),
+                        log_slice_id,
+                        num_slices,
+                    )
+                    stop_event.set()
+                    break
+
+        except NotFoundError:
+            os_logger.warning(
+                "[Slice %s/%s] PIT ID %s not found or expired, "
```

Comment:
can we somehow distinguish between expired and not found PITs?
And in a similar vain is it possible to see if the index was not found? It feels like this catch is pretty broad for some very different issues that would deserve their own handling

Reply:
Fair point. I have reworked the errors and made them more specific in their messaging.